Mlem/App/Views/Shared/Images/Helpers/AnimationControlLayer.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AnimationControlLayer.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-12-06.
+//
+
+import SwiftUICore
+
+private struct AnimationControlLayer: ViewModifier {
+    @Binding var animating: Bool
+    var muted: Binding<Bool>?
+    
+    func body(content: Content) -> some View {
+        content
+            .overlay {
+                if animating {
+                    Color.clear.contentShape(.rect)
+                        .onTapGesture {
+                            animating = false
+                        }
+                } else {
+                    PlayButton(postSize: .large)
+                        .onTapGesture {
+                            animating = true
+                        }
+                }
+            }
+            .overlay(alignment: .topTrailing) {
+                if let muted {
+                    Image(systemName: muted.wrappedValue ? Icons.muted : Icons.unmuted)
+                        .resizable()
+                        .scaledToFit()
+                        .frame(width: 15, height: 15)
+                        .padding(5)
+                        .background {
+                            Circle().foregroundStyle(.ultraThinMaterial)
+                        }
```

Comment:
*nit* this is slightly cleaner imo

```suggestion
                        .background(.ultraThinMaterial, in: .circle)
```